### ISSUE TITLE
IDEA 2023.1 Support

### ISSUE THREAD

**Issue by dvcrn** [KronicDeth/intellij-elixir#3080]
Continuation from https://github.com/KronicDeth/intellij-elixir/issues/3074#issuecomment-1439595156 since that didn't really belong into that thread... 

Having never touched Kotlin before, I tried my luck to see if I could fix the build errors:

```patch
diff --git a/.tool-versions b/.tool-versions
index bd3da7d1b..f653c6234 100644
--- a/.tool-versions
+++ b/.tool-versions
@@ -1,3 +1,3 @@
-elixir 1.13.4-otp-24
-erlang 24.3.4.6
+elixir 1.14.3-otp-25
+erlang 25.2.3
 java openjdk-17.0.2
diff --git a/build.gradle b/build.gradle
index e56e3dbcb..82173bf50 100644
--- a/build.gradle
+++ b/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-    id 'org.jetbrains.intellij' version "1.10.0"
+    id 'org.jetbrains.intellij' version "1.13.0"
     id "org.jetbrains.kotlin.jvm" version "1.7.20"
     id 'de.undercouch.download' version "4.1.2"
 }
@@ -70,7 +70,7 @@ allprojects {
         pluginDescription.set(bodyInnerHTML("resources/META-INF/description.html"))
 
         sinceBuild = "223.7571.182"
-        untilBuild = "223.*"
+        untilBuild = "231.*"
     }
 
     publishPlugin {
@@ -84,7 +84,7 @@ allprojects {
     }
 
     runPluginVerifier {
-        ideVersions = ["2022.3"]
+        ideVersions = ["2023.1"]
     }
 }
 
diff --git a/gradle.properties b/gradle.properties
index e12203a66..ddaf92fb3 100644
--- a/gradle.properties
+++ b/gradle.properties
@@ -2,11 +2,11 @@
 # https://www.jetbrains.com/intellij-repository/releases
 # https://www.jetbrains.com/intellij-repository/snapshots
 baseVersion=14.0.1
-ideaVersion=2022.3
+ideaVersion=231.7665-EAP-CANDIDATE-SNAPSHOT
 # MUST stay at 1.8 for JPS (Build/Compile Project) compatibility even if JRE/JBR is newer
 javaVersion=17
 javaTargetVersion=17
 sources=true
-elixirVersion=1.13.4
+elixirVersion=1.14.3
 publishChannels=canary
 org.gradle.jvmargs=-Xmx4096m
diff --git a/src/org/elixir_lang/code_insight/line_marker_provider/Implementation.kt b/src/org/elixir_lang/code_insight/line_marker_provider/Implementation.kt
index 7ee6f1681..a2fc91ebd 100644
--- a/src/org/elixir_lang/code_insight/line_marker_provider/Implementation.kt
+++ b/src/org/elixir_lang/code_insight/line_marker_provider/Implementation.kt
@@ -95,8 +95,8 @@ class Implementation : LineMarkerProvider {
         ) {
 
         override fun createGutterIconRenderer(
-            pointers: NotNullLazyValue<List<SmartPsiElementPointer<*>>>,
-            renderer: Computable<PsiElementListCellRenderer<*>>,
+            pointers: NotNullLazyValue<out MutableList<SmartPsiElementPointer<*>>>,
+            renderer: Computable<out PsiElementListCellRenderer<*>>,
             empty: Boolean,
             navigationHandler: GutterIconNavigationHandler<PsiElement>?
         ): NavigationGutterIconRenderer {
@@ -116,8 +116,8 @@ class Implementation : LineMarkerProvider {
     private class ProtocolsNavigationGutterIconRenderer(
         popupTitle: String?,
         emptyText: String?,
-        pointers: NotNullLazyValue<List<SmartPsiElementPointer<*>>>,
-        cellRenderer: Computable<PsiElementListCellRenderer<*>>,
+        pointers: NotNullLazyValue<out List<SmartPsiElementPointer<*>>>,
+        cellRenderer: Computable<out PsiElementListCellRenderer<*>>,
         private val alignment: Alignment,
         private val icon: Icon,
         private val tooltipText: String?,
diff --git a/src/org/elixir_lang/code_insight/line_marker_provider/Protocol.kt b/src/org/elixir_lang/code_insight/line_marker_provider/Protocol.kt
index 5e1365871..eeeaf99a4 100644
--- a/src/org/elixir_lang/code_insight/line_marker_provider/Protocol.kt
+++ b/src/org/elixir_lang/code_insight/line_marker_provider/Protocol.kt
@@ -95,12 +95,7 @@ class Protocol : LineMarkerProvider {
             PSI_GOTO_RELATED_ITEM_PROVIDER
         ) {
 
-        override fun createGutterIconRenderer(
-            pointers: NotNullLazyValue<List<SmartPsiElementPointer<*>>>,
-            renderer: Computable<PsiElementListCellRenderer<*>>,
-            empty: Boolean,
-            navigationHandler: GutterIconNavigationHandler<PsiElement>?
-        ): NavigationGutterIconRenderer {
+        override fun createGutterIconRenderer(pointers: NotNullLazyValue<out MutableList<SmartPsiElementPointer<*>>>, renderer: Computable<out PsiElementListCellRenderer<*>>, empty: Boolean, navigationHandler: GutterIconNavigationHandler<PsiElement>?): NavigationGutterIconRenderer {
             return ImplsNavigationGutterIconRenderer(
                 popupTitle = myPopupTitle,
                 emptyText = myEmptyText,
@@ -115,14 +110,14 @@ class Protocol : LineMarkerProvider {
     }
 
     private class ImplsNavigationGutterIconRenderer(
-        popupTitle: String?,
-        emptyText: String?,
-        pointers: NotNullLazyValue<List<SmartPsiElementPointer<*>>>,
-        cellRenderer: Computable<PsiElementListCellRenderer<*>>,
-        private val alignment: Alignment,
-        private val icon: Icon,
-        private val tooltipText: String?,
-        private val empty: Boolean
+            popupTitle: String?,
+            emptyText: String?,
+            pointers: NotNullLazyValue<out List<SmartPsiElementPointer<*>>>,
+            cellRenderer: Computable<out PsiElementListCellRenderer<*>>,
+            private val alignment: Alignment,
+            private val icon: Icon,
+            private val tooltipText: String?,
+            private val empty: Boolean
     ) : NavigationGutterIconRenderer(popupTitle, emptyText, cellRenderer, pointers) {
         override fun isNavigateAction(): Boolean = !empty
         override fun getIcon(): Icon = icon
diff --git a/src/org/elixir_lang/distillery/Configuration.kt b/src/org/elixir_lang/distillery/Configuration.kt
index 21af95ae0..98585bbaa 100644
--- a/src/org/elixir_lang/distillery/Configuration.kt
+++ b/src/org/elixir_lang/distillery/Configuration.kt
@@ -21,8 +21,10 @@ import org.elixir_lang.distillery.configuration.editor.CodeLoadingMode
 import org.elixir_lang.run.*
 import org.jdom.Element
 import java.io.File
+import java.nio.file.LinkOption
 import java.nio.file.Path
 import java.nio.file.Paths
+import kotlin.io.path.exists
 
 class Configuration(name: String, project: Project, configurationFactory: ConfigurationFactory) :
     org.elixir_lang.run.Configuration(name, project, configurationFactory),
@@ -235,7 +237,7 @@ class Configuration(name: String, project: Project, configurationFactory: Config
                 ?.let { release ->
                     val vmArgsPath = releasesPath.resolve(release).resolve("vm.args")
 
-                    if (vmArgsPath.exists()) {
+                    if (vmArgsPath.exists(LinkOption.NOFOLLOW_LINKS)) {
                         vmArgsPath.toString()
                     } else {
                         null
diff --git a/src/org/elixir_lang/mix/project/DirectoryConfigurator.kt b/src/org/elixir_lang/mix/project/DirectoryConfigurator.kt
index 3874a4ed8..47c36703e 100644
--- a/src/org/elixir_lang/mix/project/DirectoryConfigurator.kt
+++ b/src/org/elixir_lang/mix/project/DirectoryConfigurator.kt
@@ -26,8 +26,10 @@ import org.elixir_lang.DepsWatcher
 import org.elixir_lang.Facet
 import org.elixir_lang.mix.Project.addFolders
 import org.elixir_lang.mix.Watcher
+import java.nio.file.LinkOption
 import java.nio.file.Path
 import java.nio.file.Paths
+import kotlin.io.path.exists
 
 /**
  * Used in Small IDEs like Rubymine that don't support [OpenProcessor].
@@ -109,7 +111,7 @@ class DirectoryConfigurator : com.intellij.platform.DirectoryProjectConfigurator
     private suspend fun newProject(otpApp: OtpApp): Project? {
         val projectDir = Paths.get(FileUtil.toSystemDependentName(otpApp.root.path), Project.DIRECTORY_STORE_FOLDER)
 
-        return if (projectDir.exists()) {
+        return if (projectDir.exists(LinkOption.NOFOLLOW_LINKS)) {
             null
         } else {
             val path = otpApp.root.path.let { Paths.get(it) }
```

It's working a bit better now but still very buggy, so I think I'll have to leave that for you to handle. Some stuff I noticed that isn't working: 
- Gutter errors / no errors are being reported whatsoever, even after running dialyzer (though are in the inspections result window)
- After importing a project from "existing sources", the project does not auto-open. When manually opening, it is in a weird state where most files/packages aren't displayed in the sidebar 
- Code format isn't working

**Comment by ViseLuca** [comment]
https://github.com/KronicDeth/intellij-elixir/pull/3146 I've opened a PR to support 231.*

**Comment by KronicDeth** [comment]
Closed by#3146
